backend/gddp/src/main/scala/Gddp.scala

```diff
@@ -120,8 +121,14 @@ object Gddp extends LazyLogging {
         case None => (days: Long) => SpaceTimeKey(0, 0, days)
       }
 
+    val partitionCount: Int =
+      numPartitions match {
+        case i: Integer => i
+        case null => sc.defaultMinPartitions
```

Comment:
I've always used `defaultParallelism` for this purpose. I'm not familiar with `defaultMinPartitions` but the docs say:
"Notice that we use math.min so the "defaultMinPartitions" cannot be higher than 2." I'm not sure when that is helpful.